camel-k-loader-yaml/camel-k-loader-yaml-common/src/main/java/org/apache/camel/k/loader/yaml/parser/LoadBalanceStepParser.java

```diff
@@ -0,0 +1,158 @@
+    @Override
+    public ProcessorDefinition<?> toProcessor(Context context) {
+        Definition definition = context.node(Definition.class);
+
+        return StepParserSupport.convertSteps(
+            context,
+            definition,
+            definition.steps
+        );
+    }
+
+    @YAMLNodeDefinition(reifiers = LoadBalanceReifier.class)
+    public static final class Definition extends LoadBalanceDefinition {
+        public List<Step> steps;
+
+        @JsonAlias({"load-balancer-type", "type"})
+        @JsonTypeInfo(
+            use = JsonTypeInfo.Id.NAME,
+            include = JsonTypeInfo.As.WRAPPER_OBJECT
+        )
+        @Override
+        public void setLoadBalancerType(LoadBalancerDefinition loadbalancer) {
+            super.setLoadBalancerType(loadbalancer);
+        }
+
+        @Override
+        public LoadBalancerDefinition getLoadBalancerType() {
+            return super.getLoadBalancerType();
+        }
+
+        @JsonAlias("random")
+        public void setRandom(RandomLoadBalancerDefinition definition) {
+            if (getLoadBalancerType() != null) {
+                throw new IllegalArgumentException("A load-balancer has already been set");
+            }
+            setLoadBalancerType(definition);;
+        }
+
+        @JsonAlias({"customLoadBalancer", "custom"})
+        public  void setCustomLoadBalancer(CustomLoadBalancerDefinition definition) {
+            if (getLoadBalancerType() != null) {
+                throw new IllegalArgumentException("A load-balancer has already been set");
+            }
+            setLoadBalancerType(definition);;
+        }
+
+        @JsonAlias("failover")
+        public  void setFailover(FailoverLoadBalancerDefinition definition) {
+            if (getLoadBalancerType() != null) {
+                throw new IllegalArgumentException("A load-balancer has already been set");
+            }
+            setLoadBalancerType(definition);;
+        }
+
+        @JsonAlias("sticky")
+        public  void setSticky(Sticky definition) {
+            if (getLoadBalancerType() != null) {
+                throw new IllegalArgumentException("A load-balancer has already been set");
+            }
+            setLoadBalancerType(definition);;
+        }
+
+        @JsonAlias("topic")
+        public  void setTopic(TopicLoadBalancerDefinition definition) {
+            if (getLoadBalancerType() != null) {
+                throw new IllegalArgumentException("A load-balancer has already been set");
+            }
+            setLoadBalancerType(definition);;
+        }
+
+        @JsonAlias("weighted")
+        public  void setWeighted(WeightedLoadBalancerDefinition definition) {
+            if (getLoadBalancerType() != null) {
+                throw new IllegalArgumentException("A load-balancer has already been set");
+            }
+            setLoadBalancerType(definition);;
+        }
+
+        @JsonAlias("roundRobin")
```

Comment:
Just wonder if the alias should also have the dash variant? round-robin?

Reply:
jackson should be able to handle both the cases but I will add a test